fetcher.go

```diff
@@ -29,11 +35,18 @@ type FetchResult struct {
 	LastBlockLink ipld.Link
 }
 
-func NewFetcher(exchange *bitswap.Bitswap) Fetcher {
-	return Fetcher{exchange: exchange}
+func NewFetcherConfig(blockService blockservice.BlockService) FetcherConfig {
+	return FetcherConfig{blockService: blockService}
+}
+
+func (fc FetcherConfig) NewSession(ctx context.Context) *Fetcher {
+	return &Fetcher{
+		ctx:         ctx,
+		blockGetter: blockservice.NewSession(ctx, fc.blockService),
+	}
 }
 
-func (f Fetcher) Block(ctx context.Context, c cid.Cid) (ipld.Node, error) {
+func (f *Fetcher) Block(ctx context.Context, c cid.Cid) (ipld.Node, error) {
 	nb := basicnode.Prototype.Any.NewBuilder()
```

Comment:
can we allow the caller to pass in a NodeAssembler, rather than putting it into a basic node in the fetcher?

(or we can allow the caller to pass in an `ipld.Link` that they have constructed around a Cid, and then see if the link implements a [`ReferencedType`](https://github.com/ipld/go-ipld-prime/blob/master/schema/schema2/type_link.go#L39-L46) method to allow smarter construction of the destination node)

Reply:
I like the second suggestion!

Reply:
we should require ipld.links at the top level anyway.